velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -316,4 +316,33 @@ struct ArraySumFunction {
   }
 };
 
+template <typename TExecCtx, typename T>
+struct ArrayHasDuplicatesFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecCtx);
+
+  FOLLY_ALWAYS_INLINE void call(
+      bool& out,
+      const arg_type<velox::Array<T>>& inputArray) {
+    folly::F14FastSet<arg_type<T>> uniqSet;
+    int16_t numNulls = 0;
+    out = false;
+    for (const auto& item : inputArray) {
+      if (item.has_value()) {
+        if (uniqSet.find(item.value()) == uniqSet.end()) {
+          uniqSet.insert(item.value());
```

Comment:
You can call insert unconditionally and check the result to determine if the item is a duplicate or not.

```
if (!uniqSet.insert(item.value().second) {
   // Found duplicate.
}
```

Reply:
Fixed